demo/rust/micdemo/src/main.rs

```diff
@@ -0,0 +1,275 @@
+/*
+    Copyright 2021 Picovoice Inc.
+
+    You may not use this file except in compliance with the license. A copy of the license is located in the "LICENSE"
+    file accompanying this source.
+
+    Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+    an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+    specific language governing permissions and limitations under the License.
+*/
+
+use chrono::prelude::*;
+use clap::{App, Arg, ArgGroup};
+use ctrlc;
+use hound;
+use miniaudio;
+use rhino::{BuiltinContexts, Rhino, RhinoBuilder};
+use std::collections::VecDeque;
+use std::sync::atomic::{AtomicBool, Ordering};
+use std::sync::{Arc, Mutex};
+
+static LISTENING: AtomicBool = AtomicBool::new(false);
+
+fn process_audio_chunk(samples: &[i16], rhino: &Rhino, buffer: &mut VecDeque<i16>) {
+    buffer.extend(samples.iter());
+
+    while buffer.len() >= rhino.frame_length() as usize && LISTENING.load(Ordering::SeqCst) {
+        let frame: Vec<i16> = buffer.drain(..rhino.frame_length() as usize).collect();
+        let is_finalized = rhino.process(&frame).unwrap();
+        if is_finalized {
+            let inference = rhino.get_inference().unwrap();
+            if inference.is_understood {
+                println!("[{}] Detected {{", Local::now().format("%F %T"));
```

Comment:
maybe kick the json print down to a new line so that the structure is nice and clear